### ISSUE TITLE
Slow table scans on DefinedValue relationships

### ISSUE THREAD

**Issue by dcs619** [SparkDevNetwork/Rock#1588]
I was removing extra entries from a DefinedType and noticed it took a crazy amount of time ( almost 10 seconds for one entry).
```
DELETE FROM DefinedValue WHERE ID in (
	SELECT TOP 1 ID FROM Definedvalue WHERE Value LIKE 'http://localhost:4%'
)
```
Looking at the [query plan](https://www.dropbox.com/s/8dn1lfz4lprrfnt/Screenshot%202016-06-22%2011.47.22.png?dl=0), there's missing indexes for the following tables:
- [Attendance].SearchTypeValueId
- [Attendance].QualifierValueId
- [GroupTypeLocationType].LocationTypeValueId
- [org_sparkdevnetwork_Documentation_BookVersion].VersionDefinedValueId

I realize my machine is underpowered compared to a server, and DefinedValue touches a lot of things, but those missing indexes make up 30% of the query cost.  [Attendance] in particular will only grow by orders of magnitude. @azturner I can add a PR for a migration if you want.

On a related note, there's [a recommendation](https://www.brentozar.com/archive/2011/09/indexing-for-deletes/) to add a non-clustered index to Primary Keys, although I don't think that's the root issue here.

**Comment by azturner** [comment]
Yeah, by default EF adds indexes for any foreign keys when creating migrations for new tables, but those were probably removed at some point to try and improve query performance, not realizing impact it would have on the defined value delete (attendance table had a lot of indexes). We'll add them back (it's hard to manage pull requests that contain migrations).

**Comment by azturner** [comment]
Added indexes to core tables here: https://github.com/SparkDevNetwork/Rock/blob/f29c165382e741ebabab94acad244e5cadf6e6d0/Rock.Migrations/Migrations/201606231511599_RegistrationWorkflow_Indexes.sql

**Comment by dcs619** [comment]
Fixed in f29c165382e741ebabab94acad244e5cadf6e6d0
